File: todo-project/src/main/java/kotlinstudy/todoproject/todo/api/model/TodoResponse.java
package kotlinstudy.todoproject.todo.api.model;

import kotlinstudy.todoproject.todo.domain.Todo;
import lombok.Builder;
import lombok.Data;
import org.springframework.util.Assert;

import java.time.LocalDateTime;

@Data
@Builder
public class TodoResponse {

    private Long id;

    private String title;

    private String description;

    private Boolean done;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    public static TodoResponse of(Todo todo) {
        Assert.notNull(todo, "Todo is null");

        return TodoResponse.builder()
            .id(todo.getId())
            .title(todo.getTitle())
            .description(todo.getDescription())
            .done(todo.getDone())
            .createdAt(todo.getCreatedAt())
            .updatedAt(todo.getUpdatedAt())
            .build();
    }

}

File: todo-project/src/main/java/kotlinstudy/todoproject/todo/api/model/TodoResponse.kt
package kotlinstudy.todoproject.todo.api.model

import kotlinstudy.todoproject.todo.domain.Todo
import java.time.LocalDateTime

data class TodoResponse(
    val id: Long,
    val title: String,
    val description: String,
    val done: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime?,
) {
    companion object {
        @JvmStatic
        fun of(todo: Todo?) : TodoResponse {
            checkNotNull(todo) {"Todo is null"}
            checkNotNull(todo.id) {"Todo id is null"}

            return TodoResponse(
                id = todo.id,
                title = todo.title,
                description = todo.description,
                done = todo.done,
                createdAt = todo.createdAt,
                updatedAt = todo.updatedAt
            )
        }
    }
}
